Fix #1695, Fix #1720, Fix #1719: Removes Onboarding agreement checkbox & Updates UI.

Brandon-T opened the pull request:
<!-- *Thank you for submitting a pull request, your contributions are greatly appreciated!* -->

## Summary of Changes
Removes the onboarding terms of agreement checkbox and the logic to dim the agreement button.
Updates insets to match design.
Added new sub-title label on the agreement screen.

This pull request fixes issue #1695
This pull request fixes issue #1720 
This pull request fixes issue #1719

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
<!-- Any useful notes explaining how best to test and verify. -->


## Screenshots:
<!-- If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here. -->


## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Fix https://github.com/brave/brave-ios/issues/1695 by removing the agreement checkbox.

kylehickinson reviewed (changes requested):
There's an extra step in onboarding in this implementation:

As per flow here: https://github.com/brave/brave-ios/issues/1695

The initial "Brave Rewards" screen that used to be before the checkbox screen is gone now. Now only the one with the ToS link exists (basically just removing a screen from the flow)

This also means that "cancel" is no longer a valid button, and it will go to "Skip" the same as the other screens

Brandon-T committed:
Big updates to rewards where we changed the flow around and also the logic of what shows when.
Updated the progress state because when you click an ad, it never updated and so when the browser controller showed again, it would popup again to do the entire onboarding (infinitely unless you press skip).
Now everything is perfect.

Brandon-T commented:
Also forgot to mention, cancel no longer goes "BACKWARDS" which changed quite a bit. It always goes forwards now that it's skip and triggers the 60-day flow.

kylehickinson commented on Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementViewController.swift:
```diff
@@ -33,22 +44,23 @@ class OnboardingRewardsAgreementViewController: OnboardingViewController {
     }
     
     override func continueTapped() {
+        rewards?.ads.isEnabled = true
```
Which continue button is this? Is this this screen? 
 
![Simulator Screen Shot - iPhone 11 Pro - 2019-10-17 at 11 34 02](https://user-images.githubusercontent.com/529104/67026253-62ee5e80-f0d5-11e9-97fa-8346addfb976.png)

Brandon-T replied:
@kylehickinson 

Correct. Honestly I purposely changed the name to "continue" because this one button has had it's fair share of changes lol.. "ShowMe", "Join", "Turn On", "Continue", etc.. I just decided "continue" will be generic enough that if I change the title it just means it continues to the next screen.

If we want, I can change it's name to "turnOnButton" or something.

kylehickinson replied:
The main issue is you can't enable ads like this anymore. It will be enabled automatically based on if your region is supported when you create the wallet. The only time you have to set this to true is when you're turning on ads for existing users who have enabled rewards already when their region was unsupported and now their region is supported

kylehickinson replied:
@Brandon-T ☝️

Brandon-T replied:
Fixed.

Brandon-T committed:
Fixed constraints being broken by "1" point and made it more generic to work on any device.

Brandon-T committed:
Fix: ##1719
This centres all confirmation buttons on the onboarding screen.

Brandon-T committed:
Only enable ads for existing users that had region unavailable and it's now supported.

kylehickinson commented on Client/Frontend/Browser/Onboarding/OnboardingNavigationController.swift:
```diff
@@ -55,8 +55,8 @@ class OnboardingNavigationController: UINavigationController {
-            case .existingUserRewardsOff: return BraveAds.isCurrentRegionSupported() ? [.rewardsInfo, .rewardsAgreement, .adsCountdown] : []
+            case .newUser: return BraveAds.isCurrentRegionSupported() ? [.searchEnginePicker, .shieldsInfo, .rewardsAgreement, .adsCountdown] : [.searchEnginePicker, .shieldsInfo, .rewardsAgreement]
+            case .existingUserRewardsOff: return BraveAds.isCurrentRegionSupported() ? [.rewardsAgreement, .adsCountdown] : []
             case .existingUserRewardsOn: return BraveAds.isCurrentRegionSupported() ? [.rewardsInfo, .adsCountdown] : []
```
Can we change `rewardsInfo` to something that describes that we're just turning on ads for the user here? Since it looks like `OnboardingRewardsViewController` is only for this one flow now its a bit confusing to figure out which screen belongs to that this flow.

kylehickinson reviewed (changes requested):
Just one minor last thing for future us. Then should be good to go

Brandon-T committed:
Changed name as requested. Can't think of other names :S
